packages/auth0_api_python/src/auth0_api_python/api_client.py

```diff
@@ -120,9 +297,252 @@ async def verify_access_token(
+            raise InvalidDpopProofError("DPoP Proof iat is not recent enough")
+
+        if claims["htm"] != http_method:
+            raise InvalidDpopProofError("DPoP Proof htm mismatch")
+
+        if normalize_url_for_htu(claims["htu"]) != normalize_url_for_htu(http_url):
+            raise InvalidDpopProofError("DPoP Proof htu mismatch")
+
+        if claims["ath"] != sha256_base64url(access_token):
+            raise InvalidDpopProofError("DPoP Proof ath mismatch")
+
+        return claims
+
+    # ===== Private Methods =====
+
+    async def _discover(self) -> dict[str, Any]:
+        """Lazy-load OIDC discovery metadata."""
+        if self._metadata is None:
+            self._metadata = await fetch_oidc_metadata(
+                domain=self.options.domain,
+                custom_fetch=self.options.custom_fetch
+            )
+        return self._metadata
+
+    async def _load_jwks(self) -> dict[str, Any]:
+        """Fetches and caches JWKS data from the OIDC metadata."""
+        if self._jwks_data is None:
+            metadata = await self._discover()
+            jwks_uri = metadata["jwks_uri"]
+            self._jwks_data = await fetch_jwks(
+                jwks_uri=jwks_uri,
+                custom_fetch=self.options.custom_fetch
+            )
+        return self._jwks_data
+
+    def _validate_claims_presence(
+        self,
+        claims: dict[str, Any],
+        required_claims: list[str]
+    ) -> None:
+        """
+        Validates that all required claims are present in the claims dict.
+
+        Args:
+            claims: The claims dictionary to validate
+            required_claims: List of claim names that must be present
+
+        Raises:
+            InvalidDpopProofError: If any required claim is missing
+        """
+        missing_claims = []
+
+        for claim in required_claims:
+            if claim not in claims:
+                missing_claims.append(claim)
+
+        if missing_claims:
+            if len(missing_claims) == 1:
+                error_message = f"Missing required claim: {missing_claims[0]}"
+            else:
+                error_message = f"Missing required claims: {', '.join(missing_claims)}"
+
+            raise InvalidDpopProofError(error_message)
+
+    def _prepare_error(self, error: BaseAuthError, auth_scheme: Optional[str] = None) -> BaseAuthError:
+        """
+        Prepare an error with WWW-Authenticate headers based on error type and context.
+
+        Args:
+            error: The error to prepare
+            auth_scheme: The authentication scheme that was used ("bearer" or "dpop")
+        """
+        error_code = error.get_error_code()
+        error_description = error.get_error_description()
+
+        www_auth_headers = self._build_www_authenticate(
+            error_code=error_code if error_code != "unauthorized" else None,
+            error_description=error_description if error_code != "unauthorized" else None,
```

Comment:
This is being checked in two different places; here and inside `_build_www_authenticate`.

Reply:
I mean, `if error_code != "unauthorized"`.